utils/keys.go

```diff
@@ -0,0 +1,115 @@
+package utils
+
+import (
+	"encoding/pem"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// simple interface for the one function we need from the Storage interface
+type exportStore interface {
+	Get(string) ([]byte, error)
+	ListFiles() []string
+}
+
+type importStore interface {
+	Set(string, []byte) error
+}
+
+// ExportKeysByGUN exports all keys filtered to a GUN
+func ExportKeysByGUN(to io.Writer, s exportStore, gun string) error {
+	keys := s.ListFiles()
+	for _, k := range keys {
+		dir := filepath.Dir(k)
+		if dir == gun { // must be full GUN match
+			if err := ExportKeys(to, s, k); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportKeysByID exports all keys matching the given ID
+func ExportKeysByID(to io.Writer, s exportStore, ids []string) error {
+	want := make(map[string]struct{})
+	for _, id := range ids {
+		want[id] = struct{}{}
+	}
+	keys := s.ListFiles()
+	for _, k := range keys {
+		id := filepath.Base(k)
+		if _, ok := want[id]; ok {
+			if err := ExportKeys(to, s, k); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportKeys copies a key from the store to the io.Writer
+func ExportKeys(to io.Writer, s exportStore, from string) error {
+	// get PEM block
+	k, err := s.Get(from)
+	if err != nil {
+		return err
+	}
+
+	// parse PEM blocks if there are more than one
```

Comment:
I didn't have a good use case up front but I wanted to future proof. It seems unlikely but maybe we'll use additional PEM blocks for metadata or something like that and this will export and import them all correctly.